qoqo/src/operations/three_qubit_gate_operations.rs

```diff
@@ -0,0 +1,84 @@
+// Copyright © 2021-2023 HQS Quantum Simulations GmbH. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the
+// License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing permissions and
+// limitations under the License.
+
+use num_complex::Complex64;
+use numpy::{PyArray2, ToPyArray};
+
+use pyo3::exceptions::{PyRuntimeError, PyValueError};
+use pyo3::prelude::*;
+use pyo3::types::PySet;
+
+use qoqo_calculator::CalculatorFloat;
+use qoqo_calculator_pyo3::{convert_into_calculator_float, CalculatorFloatWrapper};
+
+use roqoqo::operations::*;
+
+use std::collections::HashMap;
+
+use crate::CircuitWrapper;
+
+use qoqo_macros::*;
+
+#[allow(clippy::upper_case_acronyms)]
+#[wrap(Operate, OperateThreeQubit, OperateGate, OperateThreeQubitGate)]
+#[derive(Eq)]
```

Comment:
Why has this been added only here?

Reply:
ControlledPauliZ derives it so I thought it made sense to derive it for the double-controlled version as well